File: flutter/auth/lib/dtos.dart
/* Options:
Date: 2020-10-05 15:05:44
Version: 5.93
Tip: To override a DTO option, remove "//" prefix before updating
BaseUrl: https://dev.servicestack.com:5001

//GlobalNamespace: 
//AddServiceStackTypes: True
//AddResponseStatus: False
//AddImplicitVersion: 
//AddDescriptionAsComments: True
//IncludeTypes: 
//ExcludeTypes: 
//DefaultImports: package:servicestack/servicestack.dart
*/

import 'package:servicestack/servicestack.dart';

abstract class IAuthTokens
{
    String provider;
    String userId;
    String accessToken;
    String accessTokenSecret;
    String refreshToken;
    DateTime refreshTokenExpiry;
    String requestToken;
    String requestTokenSecret;
    Map<String,String> items;
}

// @DataContract
class AuthUserSession implements IConvertible
{
    // @DataMember(Order=1)
    String referrerUrl;

    // @DataMember(Order=2)
    String id;

    // @DataMember(Order=3)
    String userAuthId;

    // @DataMember(Order=4)
    String userAuthName;

    // @DataMember(Order=5)
    String userName;

    // @DataMember(Order=6)
    String twitterUserId;

    // @DataMember(Order=7)
    String twitterScreenName;

    // @DataMember(Order=8)
    String facebookUserId;

    // @DataMember(Order=9)
    String facebookUserName;

    // @DataMember(Order=10)
    String firstName;

    // @DataMember(Order=11)
    String lastName;

    // @DataMember(Order=12)
    String displayName;

    // @DataMember(Order=13)
    String company;

    // @DataMember(Order=14)
    String email;

    // @DataMember(Order=15)
    String primaryEmail;

    // @DataMember(Order=16)
    String phoneNumber;

    // @DataMember(Order=17)
    DateTime birthDate;

    // @DataMember(Order=18)
    String birthDateRaw;

    // @DataMember(Order=19)
    String address;

    // @DataMember(Order=20)
    String address2;

    // @DataMember(Order=21)
    String city;

    // @DataMember(Order=22)
    String state;

    // @DataMember(Order=23)
    String country;

    // @DataMember(Order=24)
    String culture;

    // @DataMember(Order=25)
    String fullName;

    // @DataMember(Order=26)
    String gender;

    // @DataMember(Order=27)
    String language;

    // @DataMember(Order=28)
    String mailAddress;

    // @DataMember(Order=29)
    String nickname;

    // @DataMember(Order=30)
    String postalCode;

    // @DataMember(Order=31)
    String timeZone;

    // @DataMember(Order=32)
    String requestTokenSecret;

    // @DataMember(Order=33)
    DateTime createdAt;

    // @DataMember(Order=34)
    DateTime lastModified;

    // @DataMember(Order=35)
    List<String> roles;

    // @DataMember(Order=36)
    List<String> permissions;

    // @DataMember(Order=37)
    bool isAuthenticated;

    // @DataMember(Order=38)
    bool fromToken;

    // @DataMember(Order=39)
    String profileUrl;

    // @DataMember(Order=40)
    String sequence;

    // @DataMember(Order=41)
    int tag;

    // @DataMember(Order=42)
    String authProvider;

    // @DataMember(Order=43)
    List<IAuthTokens> providerOAuthAccess;

    // @DataMember(Order=44)
    Map<String,String> meta;

    // @DataMember(Order=45)
    List<String> audiences;

    // @DataMember(Order=46)
    List<String> scopes;

    // @DataMember(Order=47)
    String dns;

    // @DataMember(Order=48)
    String rsa;

    // @DataMember(Order=49)
    String sid;

    // @DataMember(Order=50)
    String hash;

    // @DataMember(Order=51)
    String homePhone;

    // @DataMember(Order=52)
    String mobilePhone;

    // @DataMember(Order=53)
    String webpage;

    // @DataMember(Order=54)
    bool emailConfirmed;

    // @DataMember(Order=55)
    bool phoneNumberConfirmed;

    // @DataMember(Order=56)
    bool twoFactorEnabled;

    // @DataMember(Order=57)
    String securityStamp;

    // @DataMember(Order=58)
    String type;

    AuthUserSession({this.referrerUrl,this.id,this.userAuthId,this.userAuthName,this.userName,this.twitterUserId,this.twitterScreenName,this.facebookUserId,this.facebookUserName,this.firstName,this.lastName,this.displayName,this.company,this.email,this.primaryEmail,this.phoneNumber,this.birthDate,this.birthDateRaw,this.address,this.address2,this.city,this.state,this.country,this.culture,this.fullName,this.gender,this.language,this.mailAddress,this.nickname,this.postalCode,this.timeZone,this.requestTokenSecret,this.createdAt,this.lastModified,this.roles,this.permissions,this.isAuthenticated,this.fromToken,this.profileUrl,this.sequence,this.tag,this.authProvider,this.providerOAuthAccess,this.meta,this.audiences,this.scopes,this.dns,this.rsa,this.sid,this.hash,this.homePhone,this.mobilePhone,this.webpage,this.emailConfirmed,this.phoneNumberConfirmed,this.twoFactorEnabled,this.securityStamp,this.type});
    AuthUserSession.fromJson(Map<String, dynamic> json) { fromMap(json); }

    fromMap(Map<String, dynamic> json) {
        referrerUrl = json['referrerUrl'];
        id = json['id'];
        userAuthId = json['userAuthId'];
        userAuthName = json['userAuthName'];
        userName = json['userName'];
        twitterUserId = json['twitterUserId'];
        twitterScreenName = json['twitterScreenName'];
        facebookUserId = json['facebookUserId'];
        facebookUserName = json['facebookUserName'];
        firstName = json['firstName'];
        lastName = json['lastName'];
        displayName = json['displayName'];
        company = json['company'];
        email = json['email'];
        primaryEmail = json['primaryEmail'];
        phoneNumber = json['phoneNumber'];
        birthDate = JsonConverters.fromJson(json['birthDate'],'DateTime',context);
        birthDateRaw = json['birthDateRaw'];
        address = json['address'];
        address2 = json['address2'];
        city = json['city'];
        state = json['state'];
        country = json['country'];
        culture = json['culture'];
        fullName = json['fullName'];
        gender = json['gender'];
        language = json['language'];
        mailAddress = json['mailAddress'];
        nickname = json['nickname'];
        postalCode = json['postalCode'];
        timeZone = json['timeZone'];
        requestTokenSecret = json['requestTokenSecret'];
        createdAt = JsonConverters.fromJson(json['createdAt'],'DateTime',context);
        lastModified = JsonConverters.fromJson(json['lastModified'],'DateTime',context);
        roles = JsonConverters.fromJson(json['roles'],'List<String>',context);
        permissions = JsonConverters.fromJson(json['permissions'],'List<String>',context);
        isAuthenticated = json['isAuthenticated'];
        fromToken = json['fromToken'];
        profileUrl = json['profileUrl'];
        sequence = json['sequence'];
        tag = json['tag'];
        authProvider = json['authProvider'];
        providerOAuthAccess = JsonConverters.fromJson(json['providerOAuthAccess'],'List<IAuthTokens>',context);
        meta = JsonConverters.toStringMap(json['meta']);
        audiences = JsonConverters.fromJson(json['audiences'],'List<String>',context);
        scopes = JsonConverters.fromJson(json['scopes'],'List<String>',context);
        dns = json['dns'];
        rsa = json['rsa'];
        sid = json['sid'];
        hash = json['hash'];
        homePhone = json['homePhone'];
        mobilePhone = json['mobilePhone'];
        webpage = json['webpage'];
        emailConfirmed = json['emailConfirmed'];
        phoneNumberConfirmed = json['phoneNumberConfirmed'];
        twoFactorEnabled = json['twoFactorEnabled'];
        securityStamp = json['securityStamp'];
        type = json['type'];
        return this;
    }

    Map<String, dynamic> toJson() => {
        'referrerUrl': referrerUrl,
        'id': id,
        'userAuthId': userAuthId,
        'userAuthName': userAuthName,
        'userName': userName,
        'twitterUserId': twitterUserId,
        'twitterScreenName': twitterScreenName,
        'facebookUserId': facebookUserId,
        'facebookUserName': facebookUserName,
        'firstName': firstName,
        'lastName': lastName,
        'displayName': displayName,
        'company': company,
        'email': email,
        'primaryEmail': primaryEmail,
        'phoneNumber': phoneNumber,
        'birthDate': JsonConverters.toJson(birthDate,'DateTime',context),
        'birthDateRaw': birthDateRaw,
        'address': address,
        'address2': address2,
        'city': city,
        'state': state,
        'country': country,
        'culture': culture,
        'fullName': fullName,
        'gender': gender,
        'language': language,
        'mailAddress': mailAddress,
        'nickname': nickname,
        'postalCode': postalCode,
        'timeZone': timeZone,
        'requestTokenSecret': requestTokenSecret,
        'createdAt': JsonConverters.toJson(createdAt,'DateTime',context),
        'lastModified': JsonConverters.toJson(lastModified,'DateTime',context),
        'roles': JsonConverters.toJson(roles,'List<String>',context),
        'permissions': JsonConverters.toJson(permissions,'List<String>',context),
        'isAuthenticated': isAuthenticated,
        'fromToken': fromToken,
        'profileUrl': profileUrl,
        'sequence': sequence,
        'tag': tag,
        'authProvider': authProvider,
        'providerOAuthAccess': JsonConverters.toJson(providerOAuthAccess,'List<IAuthTokens>',context),
        'meta': meta,
        'audiences': JsonConverters.toJson(audiences,'List<String>',context),
        'scopes': JsonConverters.toJson(scopes,'List<String>',context),
        'dns': dns,
        'rsa': rsa,
        'sid': sid,
        'hash': hash,
        'homePhone': homePhone,
        'mobilePhone': mobilePhone,
        'webpage': webpage,
        'emailConfirmed': emailConfirmed,
        'phoneNumberConfirmed': phoneNumberConfirmed,
        'twoFactorEnabled': twoFactorEnabled,
        'securityStamp': securityStamp,
        'type': type
    };

    TypeContext context = _ctx;
}

class CustomUserSession extends AuthUserSession implements IConvertible
{
    CustomUserSession();
    CustomUserSession.fromJson(Map<String, dynamic> json) : super.fromJson(json);
    fromMap(Map<String, dynamic> json) {
        super.fromMap(json);
        return this;
    }

    Map<String, dynamic> toJson() => super.toJson();
    TypeContext context = _ctx;
}

class UserAuth implements IConvertible
{
    int id;
    String userName;
    String email;
    String primaryEmail;
    String phoneNumber;
    String firstName;
    String lastName;
    String displayName;
    String company;
    DateTime birthDate;
    String birthDateRaw;
    String address;
    String address2;
    String city;
    String state;
    String country;
    String culture;
    String fullName;
    String gender;
    String language;
    String mailAddress;
    String nickname;
    String postalCode;
    String timeZone;
    String salt;
    String passwordHash;
    String digestHa1Hash;
    List<String> roles;
    List<String> permissions;
    DateTime createdDate;
    DateTime modifiedDate;
    int invalidLoginAttempts;
    DateTime lastLoginAttempt;
    DateTime lockedDate;
    String recoveryToken;
    int refId;
    String refIdStr;
    Map<String,String> meta;

    UserAuth({this.id,this.userName,this.email,this.primaryEmail,this.phoneNumber,this.firstName,this.lastName,this.displayName,this.company,this.birthDate,this.birthDateRaw,this.address,this.address2,this.city,this.state,this.country,this.culture,this.fullName,this.gender,this.language,this.mailAddress,this.nickname,this.postalCode,this.timeZone,this.salt,this.passwordHash,this.digestHa1Hash,this.roles,this.permissions,this.createdDate,this.modifiedDate,this.invalidLoginAttempts,this.lastLoginAttempt,this.lockedDate,this.recoveryToken,this.refId,this.refIdStr,this.meta});
    UserAuth.fromJson(Map<String, dynamic> json) { fromMap(json); }

    fromMap(Map<String, dynamic> json) {
        id = json['id'];
        userName = json['userName'];
        email = json['email'];
        primaryEmail = json['primaryEmail'];
        phoneNumber = json['phoneNumber'];
        firstName = json['firstName'];
        lastName = json['lastName'];
        displayName = json['displayName'];
        company = json['company'];
        birthDate = JsonConverters.fromJson(json['birthDate'],'DateTime',context);
        birthDateRaw = json['birthDateRaw'];
        address = json['address'];
        address2 = json['address2'];
        city = json['city'];
        state = json['state'];
        country = json['country'];
        culture = json['culture'];
        fullName = json['fullName'];
        gender = json['gender'];
        language = json['language'];
        mailAddress = json['mailAddress'];
        nickname = json['nickname'];
        postalCode = json['postalCode'];
        timeZone = json['timeZone'];
        salt = json['salt'];
        passwordHash = json['passwordHash'];
        digestHa1Hash = json['digestHa1Hash'];
        roles = JsonConverters.fromJson(json['roles'],'List<String>',context);
        permissions = JsonConverters.fromJson(json['permissions'],'List<String>',context);
        createdDate = JsonConverters.fromJson(json['createdDate'],'DateTime',context);
        modifiedDate = JsonConverters.fromJson(json['modifiedDate'],'DateTime',context);
        invalidLoginAttempts = json['invalidLoginAttempts'];
        lastLoginAttempt = JsonConverters.fromJson(json['lastLoginAttempt'],'DateTime',context);
        lockedDate = JsonConverters.fromJson(json['lockedDate'],'DateTime',context);
        recoveryToken = json['recoveryToken'];
        refId = json['refId'];
        refIdStr = json['refIdStr'];
        meta = JsonConverters.toStringMap(json['meta']);
        return this;
    }

    Map<String, dynamic> toJson() => {
        'id': id,
        'userName': userName,
        'email': email,
        'primaryEmail': primaryEmail,
        'phoneNumber': phoneNumber,
        'firstName': firstName,
        'lastName': lastName,
        'displayName': displayName,
        'company': company,
        'birthDate': JsonConverters.toJson(birthDate,'DateTime',context),
        'birthDateRaw': birthDateRaw,
        'address': address,
        'address2': address2,
        'city': city,
        'state': state,
        'country': country,
        'culture': culture,
        'fullName': fullName,
        'gender': gender,
        'language': language,
        'mailAddress': mailAddress,
        'nickname': nickname,
        'postalCode': postalCode,
        'timeZone': timeZone,
        'salt': salt,
        'passwordHash': passwordHash,
        'digestHa1Hash': digestHa1Hash,
        'roles': JsonConverters.toJson(roles,'List<String>',context),
        'permissions': JsonConverters.toJson(permissions,'List<String>',context),
        'createdDate': JsonConverters.toJson(createdDate,'DateTime',context),
        'modifiedDate': JsonConverters.toJson(modifiedDate,'DateTime',context),
        'invalidLoginAttempts': invalidLoginAttempts,
        'lastLoginAttempt': JsonConverters.toJson(lastLoginAttempt,'DateTime',context),
        'lockedDate': JsonConverters.toJson(lockedDate,'DateTime',context),
        'recoveryToken': recoveryToken,
        'refId': refId,
        'refIdStr': refIdStr,
        'meta': meta
    };

    TypeContext context = _ctx;
}

class AppUser extends UserAuth implements IConvertible
{
    String profileUrl;
    String lastLoginIp;
    DateTime lastLoginDate;

    AppUser({this.profileUrl,this.lastLoginIp,this.lastLoginDate});
    AppUser.fromJson(Map<String, dynamic> json) { fromMap(json); }

    fromMap(Map<String, dynamic> json) {
        super.fromMap(json);
        profileUrl = json['profileUrl'];
        lastLoginIp = json['lastLoginIp'];
        lastLoginDate = JsonConverters.fromJson(json['lastLoginDate'],'DateTime',context);
        return this;
    }

    Map<String, dynamic> toJson() => super.toJson()..addAll({
        'profileUrl': profileUrl,
        'lastLoginIp': lastLoginIp,
        'lastLoginDate': JsonConverters.toJson(lastLoginDate,'DateTime',context)
    });

    TypeContext context = _ctx;
}

class UserAuthDetails implements IConvertible
{
    int id;
    int userAuthId;
    String provider;
    String userId;
    String userName;
    String fullName;
    String displayName;
    String firstName;
    String lastName;
    String company;
    String email;
    String phoneNumber;
    DateTime birthDate;
    String birthDateRaw;
    String address;
    String address2;
    String city;
    String state;
    String country;
    String culture;
    String gender;
    String language;
    String mailAddress;
    String nickname;
    String postalCode;
    String timeZone;
    String refreshToken;
    DateTime refreshTokenExpiry;
    String requestToken;
    String requestTokenSecret;
    Map<String,String> items;
    String accessToken;
    String accessTokenSecret;
    DateTime createdDate;
    DateTime modifiedDate;
    int refId;
    String refIdStr;
    Map<String,String> meta;

    UserAuthDetails({this.id,this.userAuthId,this.provider,this.userId,this.userName,this.fullName,this.displayName,this.firstName,this.lastName,this.company,this.email,this.phoneNumber,this.birthDate,this.birthDateRaw,this.address,this.address2,this.city,this.state,this.country,this.culture,this.gender,this.language,this.mailAddress,this.nickname,this.postalCode,this.timeZone,this.refreshToken,this.refreshTokenExpiry,this.requestToken,this.requestTokenSecret,this.items,this.accessToken,this.accessTokenSecret,this.createdDate,this.modifiedDate,this.refId,this.refIdStr,this.meta});
    UserAuthDetails.fromJson(Map<String, dynamic> json) { fromMap(json); }

    fromMap(Map<String, dynamic> json) {
        id = json['id'];
        userAuthId = json['userAuthId'];
        provider = json['provider'];
        userId = json['userId'];
        userName = json['userName'];
        fullName = json['fullName'];
        displayName = json['displayName'];
        firstName = json['firstName'];
        lastName = json['lastName'];
        company = json['company'];
        email = json['email'];
        phoneNumber = json['phoneNumber'];
        birthDate = JsonConverters.fromJson(json['birthDate'],'DateTime',context);
        birthDateRaw = json['birthDateRaw'];
        address = json['address'];
        address2 = json['address2'];
        city = json['city'];
        state = json['state'];
        country = json['country'];
        culture = json['culture'];
        gender = json['gender'];
        language = json['language'];
        mailAddress = json['mailAddress'];
        nickname = json['nickname'];
        postalCode = json['postalCode'];
        timeZone = json['timeZone'];
        refreshToken = json['refreshToken'];
        refreshTokenExpiry = JsonConverters.fromJson(json['refreshTokenExpiry'],'DateTime',context);
        requestToken = json['requestToken'];
        requestTokenSecret = json['requestTokenSecret'];
        items = JsonConverters.toStringMap(json['items']);
        accessToken = json['accessToken'];
        accessTokenSecret = json['accessTokenSecret'];
        createdDate = JsonConverters.fromJson(json['createdDate'],'DateTime',context);
        modifiedDate = JsonConverters.fromJson(json['modifiedDate'],'DateTime',context);
        refId = json['refId'];
        refIdStr = json['refIdStr'];
        meta = JsonConverters.toStringMap(json['meta']);
        return this;
    }

    Map<String, dynamic> toJson() => {
        'id': id,
        'userAuthId': userAuthId,
        'provider': provider,
        'userId': userId,
        'userName': userName,
        'fullName': fullName,
        'displayName': displayName,
        'firstName': firstName,
        'lastName': lastName,
        'company': company,
        'email': email,
        'phoneNumber': phoneNumber,
        'birthDate': JsonConverters.toJson(birthDate,'DateTime',context),
        'birthDateRaw': birthDateRaw,
        'address': address,
        'address2': address2,
        'city': city,
        'state': state,
        'country': country,
        'culture': culture,
        'gender': gender,
        'language': language,
        'mailAddress': mailAddress,
        'nickname': nickname,
        'postalCode': postalCode,
        'timeZone': timeZone,
        'refreshToken': refreshToken,
        'refreshTokenExpiry': JsonConverters.toJson(refreshTokenExpiry,'DateTime',context),
        'requestToken': requestToken,
        'requestTokenSecret': requestTokenSecret,
        'items': items,
        'accessToken': accessToken,
        'accessTokenSecret': accessTokenSecret,
        'createdDate': JsonConverters.toJson(createdDate,'DateTime',context),
        'modifiedDate': JsonConverters.toJson(modifiedDate,'DateTime',context),
        'refId': refId,
        'refIdStr': refIdStr,
        'meta': meta
    };

    TypeContext context = _ctx;
}

class HelloResponse implements IConvertible
{
    String result;

    HelloResponse({this.result});
    HelloResponse.fromJson(Map<String, dynamic> json) { fromMap(json); }

    fromMap(Map<String, dynamic> json) {
        result = json['result'];
        return this;
    }

    Map<String, dynamic> toJson() => {
        'result': result
    };

    TypeContext context = _ctx;
}

class AllUserInfoResponse implements IConvertible
{
    ResponseStatus responseStatus;
    List<CustomUserSession> userSessions;
    List<AppUser> appUsers;
    List<UserAuthDetails> userAuthDetails;

    AllUserInfoResponse({this.responseStatus,this.userSessions,this.appUsers,this.userAuthDetails});
    AllUserInfoResponse.fromJson(Map<String, dynamic> json) { fromMap(json); }

    fromMap(Map<String, dynamic> json) {
        responseStatus = JsonConverters.fromJson(json['responseStatus'],'ResponseStatus',context);
        userSessions = JsonConverters.fromJson(json['userSessions'],'List<CustomUserSession>',context);
        appUsers = JsonConverters.fromJson(json['appUsers'],'List<AppUser>',context);
        userAuthDetails = JsonConverters.fromJson(json['userAuthDetails'],'List<UserAuthDetails>',context);
        return this;
    }

    Map<String, dynamic> toJson() => {
        'responseStatus': JsonConverters.toJson(responseStatus,'ResponseStatus',context),
        'userSessions': JsonConverters.toJson(userSessions,'List<CustomUserSession>',context),
        'appUsers': JsonConverters.toJson(appUsers,'List<AppUser>',context),
        'userAuthDetails': JsonConverters.toJson(userAuthDetails,'List<UserAuthDetails>',context)
    };

    TypeContext context = _ctx;
}

// @Route("/hello")
// @Route("/hello/{Name}")
class Hello implements IReturn<HelloResponse>, IConvertible
{
    String name;

    Hello({this.name});
    Hello.fromJson(Map<String, dynamic> json) { fromMap(json); }

    fromMap(Map<String, dynamic> json) {
        name = json['name'];
        return this;
    }

    Map<String, dynamic> toJson() => {
        'name': name
    };

    createResponse() { return new HelloResponse(); }
    String getTypeName() { return "Hello"; }
    TypeContext context = _ctx;
}

// @Route("/hello/secure")
// @Route("/hello/secure/{Name}")
// @ValidateRequest(Validator="IsAuthenticated")
class HelloSecure implements IReturn<HelloResponse>, IConvertible
{
    String name;

    HelloSecure({this.name});
    HelloSecure.fromJson(Map<String, dynamic> json) { fromMap(json); }

    fromMap(Map<String, dynamic> json) {
        name = json['name'];
        return this;
    }

    Map<String, dynamic> toJson() => {
        'name': name
    };

    createResponse() { return new HelloResponse(); }
    String getTypeName() { return "HelloSecure"; }
    TypeContext context = _ctx;
}

// @Route("/users")
class AllUserInfo implements IReturn<AllUserInfoResponse>, IConvertible
{
    AllUserInfo();
    AllUserInfo.fromJson(Map<String, dynamic> json) : super();
    fromMap(Map<String, dynamic> json) {
        return this;
    }

    Map<String, dynamic> toJson() => {};
    createResponse() { return new AllUserInfoResponse(); }
    String getTypeName() { return "AllUserInfo"; }
    TypeContext context = _ctx;
}

TypeContext _ctx = new TypeContext(library: 'dev.servicestack.com', types: <String, TypeInfo> {
    'IAuthTokens': new TypeInfo(TypeOf.Interface),
    'AuthUserSession': new TypeInfo(TypeOf.Class, create:() => new AuthUserSession()),
    'List<IAuthTokens>': new TypeInfo(TypeOf.Class, create:() => new List<IAuthTokens>()),
    'CustomUserSession': new TypeInfo(TypeOf.Class, create:() => new CustomUserSession()),
    'UserAuth': new TypeInfo(TypeOf.Class, create:() => new UserAuth()),
    'AppUser': new TypeInfo(TypeOf.Class, create:() => new AppUser()),
    'UserAuthDetails': new TypeInfo(TypeOf.Class, create:() => new UserAuthDetails()),
    'HelloResponse': new TypeInfo(TypeOf.Class, create:() => new HelloResponse()),
    'AllUserInfoResponse': new TypeInfo(TypeOf.Class, create:() => new AllUserInfoResponse()),
    'List<CustomUserSession>': new TypeInfo(TypeOf.Class, create:() => new List<CustomUserSession>()),
    'List<AppUser>': new TypeInfo(TypeOf.Class, create:() => new List<AppUser>()),
    'List<UserAuthDetails>': new TypeInfo(TypeOf.Class, create:() => new List<UserAuthDetails>()),
    'Hello': new TypeInfo(TypeOf.Class, create:() => new Hello()),
    'HelloSecure': new TypeInfo(TypeOf.Class, create:() => new HelloSecure()),
    'AllUserInfo': new TypeInfo(TypeOf.Class, create:() => new AllUserInfo()),
});

File: swift/MyApp/MyApp/dtos.swift
/* Options:
Date: 2020-10-08 01:39:36
SwiftVersion: 4.0
Version: 5.9
Tip: To override a DTO option, remove "//" prefix before updating
BaseUrl: https://dev.servicestack.com:5001

//BaseClass: 
//AddModelExtensions: True
//AddServiceStackTypes: True
IncludeTypes: Hello.*,HelloSecure.*,Authenticate.*
//ExcludeTypes: 
//ExcludeGenericBaseTypes: False
//AddResponseStatus: False
//AddImplicitVersion: 
//AddDescriptionAsComments: True
//InitializeCollections: True
//TreatTypesAsStrings: 
//DefaultImports: Foundation,ServiceStack
*/

import Foundation
import ServiceStack

// @Route("/hello")
// @Route("/hello/{Name}")
// @DataContract
public class Hello : IReturn
{
    public typealias Return = HelloResponse

    required public init(){}
    // @DataMember(Order=1)
    public var name:String?
}

// @Route("/hello/secure")
// @Route("/hello/secure/{Name}")
// @ValidateRequest(Validator="IsAuthenticated")
public class HelloSecure : IReturn
{
    public typealias Return = HelloResponse

    required public init(){}
    public var name:String?
}

// @Route("/auth")
// @Route("/auth/{provider}")
// @DataContract
public class Authenticate : IReturn, IPost
{
    public typealias Return = AuthenticateResponse

    required public init(){}
    // @DataMember(Order=1)
    public var provider:String?

    // @DataMember(Order=2)
    public var state:String?

    // @DataMember(Order=3)
    public var oauth_token:String?

    // @DataMember(Order=4)
    public var oauth_verifier:String?

    // @DataMember(Order=5)
    public var userName:String?

    // @DataMember(Order=6)
    public var password:String?

    // @DataMember(Order=7)
    public var rememberMe:Bool?

    // @DataMember(Order=9)
    public var errorView:String?

    // @DataMember(Order=10)
    public var nonce:String?

    // @DataMember(Order=11)
    public var uri:String?

    // @DataMember(Order=12)
    public var response:String?

    // @DataMember(Order=13)
    public var qop:String?

    // @DataMember(Order=14)
    public var nc:String?

    // @DataMember(Order=15)
    public var cnonce:String?

    // @DataMember(Order=16)
    public var useTokenCookie:Bool?

    // @DataMember(Order=17)
    public var accessToken:String?

    // @DataMember(Order=18)
    public var accessTokenSecret:String?

    // @DataMember(Order=19)
    public var scope:String?

    // @DataMember(Order=20)
    public var meta:[String:String] = [:]
}

// @DataContract
public class HelloResponse
{
    required public init(){}
    // @DataMember(Order=1)
    public var result:String?
}

// @DataContract
public class AuthenticateResponse : IHasSessionId, IHasBearerToken
{
    required public init(){}
    // @DataMember(Order=1)
    public var userId:String?

    // @DataMember(Order=2)
    public var sessionId:String?

    // @DataMember(Order=3)
    public var userName:String?

    // @DataMember(Order=4)
    public var displayName:String?

    // @DataMember(Order=5)
    public var referrerUrl:String?

    // @DataMember(Order=6)
    public var bearerToken:String?

    // @DataMember(Order=7)
    public var refreshToken:String?

    // @DataMember(Order=8)
    public var profileUrl:String?

    // @DataMember(Order=9)
    public var roles:[String] = []

    // @DataMember(Order=10)
    public var permissions:[String] = []

    // @DataMember(Order=11)
    public var responseStatus:ResponseStatus?

    // @DataMember(Order=12)
    public var meta:[String:String] = [:]
}


extension Hello : JsonSerializable
{
    public static var typeName:String { return "Hello" }
    public static var metadata = Metadata.create([
            Type<Hello>.optionalProperty("name", get: { $0.name }, set: { $0.name = $1 }),
        ])
}

extension HelloSecure : JsonSerializable
{
    public static var typeName:String { return "HelloSecure" }
    public static var metadata = Metadata.create([
            Type<HelloSecure>.optionalProperty("name", get: { $0.name }, set: { $0.name = $1 }),
        ])
}

extension Authenticate : JsonSerializable
{
    public static var typeName:String { return "Authenticate" }
    public static var metadata = Metadata.create([
            Type<Authenticate>.optionalProperty("provider", get: { $0.provider }, set: { $0.provider = $1 }),
            Type<Authenticate>.optionalProperty("state", get: { $0.state }, set: { $0.state = $1 }),
            Type<Authenticate>.optionalProperty("oauth_token", get: { $0.oauth_token }, set: { $0.oauth_token = $1 }),
            Type<Authenticate>.optionalProperty("oauth_verifier", get: { $0.oauth_verifier }, set: { $0.oauth_verifier = $1 }),
            Type<Authenticate>.optionalProperty("userName", get: { $0.userName }, set: { $0.userName = $1 }),
            Type<Authenticate>.optionalProperty("password", get: { $0.password }, set: { $0.password = $1 }),
            Type<Authenticate>.optionalProperty("rememberMe", get: { $0.rememberMe }, set: { $0.rememberMe = $1 }),
            Type<Authenticate>.optionalProperty("errorView", get: { $0.errorView }, set: { $0.errorView = $1 }),
            Type<Authenticate>.optionalProperty("nonce", get: { $0.nonce }, set: { $0.nonce = $1 }),
            Type<Authenticate>.optionalProperty("uri", get: { $0.uri }, set: { $0.uri = $1 }),
            Type<Authenticate>.optionalProperty("response", get: { $0.response }, set: { $0.response = $1 }),
            Type<Authenticate>.optionalProperty("qop", get: { $0.qop }, set: { $0.qop = $1 }),
            Type<Authenticate>.optionalProperty("nc", get: { $0.nc }, set: { $0.nc = $1 }),
            Type<Authenticate>.optionalProperty("cnonce", get: { $0.cnonce }, set: { $0.cnonce = $1 }),
            Type<Authenticate>.optionalProperty("useTokenCookie", get: { $0.useTokenCookie }, set: { $0.useTokenCookie = $1 }),
            Type<Authenticate>.optionalProperty("accessToken", get: { $0.accessToken }, set: { $0.accessToken = $1 }),
            Type<Authenticate>.optionalProperty("accessTokenSecret", get: { $0.accessTokenSecret }, set: { $0.accessTokenSecret = $1 }),
            Type<Authenticate>.optionalProperty("scope", get: { $0.scope }, set: { $0.scope = $1 }),
            Type<Authenticate>.objectProperty("meta", get: { $0.meta }, set: { $0.meta = $1 }),
        ])
}

extension HelloResponse : JsonSerializable
{
    public static var typeName:String { return "HelloResponse" }
    public static var metadata = Metadata.create([
            Type<HelloResponse>.optionalProperty("result", get: { $0.result }, set: { $0.result = $1 }),
        ])
}

extension AuthenticateResponse : JsonSerializable
{
    public static var typeName:String { return "AuthenticateResponse" }
    public static var metadata = Metadata.create([
            Type<AuthenticateResponse>.optionalProperty("userId", get: { $0.userId }, set: { $0.userId = $1 }),
            Type<AuthenticateResponse>.optionalProperty("sessionId", get: { $0.sessionId }, set: { $0.sessionId = $1 }),
            Type<AuthenticateResponse>.optionalProperty("userName", get: { $0.userName }, set: { $0.userName = $1 }),
            Type<AuthenticateResponse>.optionalProperty("displayName", get: { $0.displayName }, set: { $0.displayName = $1 }),
            Type<AuthenticateResponse>.optionalProperty("referrerUrl", get: { $0.referrerUrl }, set: { $0.referrerUrl = $1 }),
            Type<AuthenticateResponse>.optionalProperty("bearerToken", get: { $0.bearerToken }, set: { $0.bearerToken = $1 }),
            Type<AuthenticateResponse>.optionalProperty("refreshToken", get: { $0.refreshToken }, set: { $0.refreshToken = $1 }),
            Type<AuthenticateResponse>.optionalProperty("profileUrl", get: { $0.profileUrl }, set: { $0.profileUrl = $1 }),
            Type<AuthenticateResponse>.arrayProperty("roles", get: { $0.roles }, set: { $0.roles = $1 }),
            Type<AuthenticateResponse>.arrayProperty("permissions", get: { $0.permissions }, set: { $0.permissions = $1 }),
            Type<AuthenticateResponse>.optionalProperty("responseStatus", get: { $0.responseStatus }, set: { $0.responseStatus = $1 }),
            Type<AuthenticateResponse>.objectProperty("meta", get: { $0.meta }, set: { $0.meta = $1 }),
        ])
}

